packages/components/src/Modal/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -39,6 +39,9 @@ export type ConfirmationModalProps = {
   title: string
   onConfirm?: () => void
   onDismiss: () => void
+  /** A callback that is triggered after the modal is opened. */
+  onAfterEnter?: () => void
```

Comment:
I've gone with `onAfterEnter` instead of `onAfterOpen` to keep in line with the convention in this component  since we have a `onAfterEnterHandler` which is then passed into `afterEnter` in the Transition component